src/npm/Prefetcher.ts

```diff
@@ -84,11 +87,32 @@ export class Prefetcher {
         return;
       }
 
-      this.#ready.push(...packages);
+      await this.queueIndex.saveObjects(
+        packages.map((pkg) => ({
+          objectID: pkg.id,
+          retries: 0,
+          pkg,
+        }))
+      );
+
+      const lastId = (await this.stateManager.get()).bootstrapLastId;
+      const pkg = packages.at(-1);
+
+      if (pkg && (!lastId || lastId < pkg.id)) {
+        await this.stateManager.save({
+          bootstrapLastId: pkg.id,
+        });
```

Comment:
Yes, I'm aware of these issues and used `.wait()` where necessary. In other places, the worst case is we occasionally process something twice, but it shouldn't happen too often. I agree Algolia is not the best fit for this (and even the queues) but didn't want to make the scope even bigger than it was at this point.

Reply:
With `IncrementFrom` I don't think the sequence resets matter much. Since that operation accepts only a single matching value, a conflict would only happen if there was a combination of the same package name and `seq` within a few minutes after the reset, which is extremely unlikely.

As for `deleteBy`, note that we don't need it on the main index with 3M records, only on the queue index where we have < 100 records, typically. Also, the field doesn't have a high cardinality - only two values are possible, one of which means "delete me". Right now, we make one such call per minute and the interval could be increased, if needed.

Reply:
Yeah I don't think it's a big issue here, just raising some points to be sure ☺️